### ISSUE TITLE
Wrong insert_id returned when using DifferentDB

### ISSUE THREAD

**Issue by NyroTheCat** [reinhardfuehricht/typo3-formhandler#15]
When `Finisher\DifferentDB` is used, the parent class won't return the correct insert_id after save() method was successfully executed. The underlying process() accesses the wrong connection resource. 

```php
// Finisher\DB.php
public function process()
{
   ...
   $this->gp['inserted_uid'] = $GLOBALS['TYPO3_DB']->sql_insert_id();
   ...
}
```

This will return the `insert_id` of the last inserted debug message instead. I've created a fix for this. Besides that i gave the `Finisher\DifferentDB` class a complete overhault.

see pull request #14

**Comment by jmverges** [comment]
closed: #14

**Comment by jmverges** [comment]
close: #14
